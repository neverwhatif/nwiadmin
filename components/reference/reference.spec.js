import React from 'react';
import { shallow } from 'enzyme';

import Reference from './index';

jest.mock('nwiadmin/components/link', () => 'Link');

describe('Reference', () => {
    const component = shallow(<Reference title="Lorem" reference="ipsum" />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(component.text()).toBe('Loremipsum');
    });
    it('should render reference', () => {
        expect(component.find('.reference').length).toBe(1);
        expect(component.find('.reference').text()).toBe('ipsum');
    });
});
