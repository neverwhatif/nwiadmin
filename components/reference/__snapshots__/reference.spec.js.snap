// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reference should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reference
    link={null}
    reference="ipsum"
    title="Lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Lorem",
        <span
          className="reference"
        >
          ipsum
        </span>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      "Lorem",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ipsum",
          "className": "reference",
        },
        "ref": null,
        "rendered": "ipsum",
        "type": "span",
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Lorem",
          <span
            className="reference"
          >
            ipsum
          </span>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        "Lorem",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ipsum",
            "className": "reference",
          },
          "ref": null,
          "rendered": "ipsum",
          "type": "span",
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
